model/lib/extensions/presentation_mailbox_extension.dart

```diff
@@ -1,8 +1,73 @@
 import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
-import 'package:model/model.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/namespace.dart';
+import 'package:model/mailbox/mailbox_state.dart';
+import 'package:model/mailbox/presentation_mailbox.dart';
+import 'package:model/mailbox/select_mode.dart';
 
 extension PresentationMailboxExtension on PresentationMailbox {
 
+  bool get isActivated => state == MailboxState.activated;
+
+  bool hasParentId() => parentId != null && parentId!.id.value.isNotEmpty;
+
+  bool hasRole() => role != null && role!.value.isNotEmpty;
+
+  bool get isDefault => hasRole();
+
+  bool get isPersonal => namespace == null || namespace == Namespace('Personal');
+
+  bool get isTeamMailboxes => !isPersonal && !hasParentId();
+
+  bool get isChildOfTeamMailboxes => !isPersonal && hasParentId();
+
+  String get countUnReadEmailsAsString {
+    if (countUnreadEmails <= 0) return '';
+    return countUnreadEmails <= 999 ? '$countUnreadEmails' : '999+';
+  }
+
+  int get countUnreadEmails => unreadEmails?.value.value.toInt() ?? 0;
+
+  int get countTotalEmails => totalEmails?.value.value.toInt() ?? 0;
+
+  bool get isSpam => role == PresentationMailbox.roleSpam;
+
+  bool get isTrash => role == PresentationMailbox.roleTrash;
+
+  bool get isDrafts => role == PresentationMailbox.roleDrafts;
+
+  bool get isTemplates => role == PresentationMailbox.roleTemplates;
+
+  bool get isSent => role == PresentationMailbox.roleSent;
+
+  bool get isOutbox => name == PresentationMailbox.lowerCaseOutboxMailboxName || role == PresentationMailbox.roleOutbox;
+
+  bool get isSubscribedMailbox => isSubscribed != null && isSubscribed?.value == true;
+
+  bool get allowedToDisplayCountOfUnreadEmails => !(isTrash || isSpam || isDrafts || isTemplates || isSent);
+
+  bool get allowedToDisplayCountOfTotalEmails => isTrash || isSpam;
+
+  String get countTotalEmailsAsString {
+    if (countTotalEmails <= 0) return '';
+    return countTotalEmails <= 999 ? '$countTotalEmails' : '999+';
+  }
+
+  String? get emailTeamMailBoxes => namespace?.value.substring(
+    (namespace?.value.indexOf('[') ?? 0) + 1,
+    namespace?.value.indexOf(']'));
```

Comment:
What if the value doesn't contains ']', is this value equal to -1, or throw an exception. do you think it safe ?

Reply:
1. Srr, Really this is not the one to review in this PR. It was handled in the previous PR. I adjusted to move it to another class to synchronize.
2. Make sure it contains 2 characters `[` and `]`. Because the team mailbox is formatted like that. Of course looking like that it's very specific :). But that's okay, that's all we need for now.